src/nci_util.c

```diff
@@ -912,6 +1040,173 @@ nci_discovery_ntf_copy(
+
+        switch (intf) {
+        case NCI_RF_INTERFACE_ISO_DEP:
+            switch (mode) {
+            case NCI_MODE_PASSIVE_POLL_A:
+            case NCI_MODE_ACTIVE_POLL_A:
+                iso_dep_poll_a = &param->iso_dep_poll_a;
+                GASSERT(!iso_dep_poll_a->t1.bytes || iso_dep_poll_a->t1.size);
+                GASSERT(!iso_dep_poll_a->t1.size || iso_dep_poll_a->t1.bytes);
+                if (iso_dep_poll_a->t1.size) {
+                    size = G_ALIGN8(size);
+                    size += G_ALIGN8(iso_dep_poll_a->t1.size);
+                }
+                copy = g_malloc0(size);
+                memcpy(copy, param, sizeof(NciActivationParam));
+                if (iso_dep_poll_a->t1.bytes) {
+                    guint8* dest = (guint8*)copy +
+                        G_ALIGN8(sizeof(NciActivationParam));
+                    memcpy(dest, iso_dep_poll_a->t1.bytes,
+                        iso_dep_poll_a->t1.size);
+                    copy->iso_dep_poll_a.t1.bytes = dest;
+                }
+                return copy;
+            case NCI_MODE_PASSIVE_POLL_B:
+                iso_dep_poll_b = &param->iso_dep_poll_b;
+                GASSERT(!iso_dep_poll_b->hlr.bytes || iso_dep_poll_b->hlr.size);
+                GASSERT(!iso_dep_poll_b->hlr.size || iso_dep_poll_b->hlr.bytes);
+                if (iso_dep_poll_b->hlr.size) {
+                    size = G_ALIGN8(size);
+                    size += G_ALIGN8(iso_dep_poll_b->hlr.size);
+                }
+                copy = g_malloc0(size);
+                memcpy(copy, param, sizeof(NciActivationParam));
+                if (iso_dep_poll_b->hlr.bytes) {
+                    guint8* dest = (guint8*)copy +
+                        G_ALIGN8(sizeof(NciActivationParam));
+                    memcpy(dest, iso_dep_poll_b->hlr.bytes,
+                        iso_dep_poll_b->hlr.size);
+                    copy->iso_dep_poll_b.hlr.bytes = dest;
+                }
+                return copy;
+            case NCI_MODE_PASSIVE_POLL_F:
+            case NCI_MODE_ACTIVE_POLL_F:
+            case NCI_MODE_PASSIVE_POLL_15693:
+            case NCI_MODE_PASSIVE_LISTEN_A:
+            case NCI_MODE_PASSIVE_LISTEN_B:
+            case NCI_MODE_PASSIVE_LISTEN_F:
+            case NCI_MODE_ACTIVE_LISTEN_A:
+            case NCI_MODE_ACTIVE_LISTEN_F:
+            case NCI_MODE_PASSIVE_LISTEN_15693:
+                break;
+            }
+            break;
+        case NCI_RF_INTERFACE_FRAME:
+            /* There are no Activation Parameters for Frame RF interface */
+            break;
+        case NCI_RF_INTERFACE_NFC_DEP:
+            switch (mode) {
+            case NCI_MODE_ACTIVE_POLL_A:
+            case NCI_MODE_ACTIVE_POLL_F:
+            case NCI_MODE_PASSIVE_POLL_A:
+            case NCI_MODE_PASSIVE_POLL_F:
+                nfc_dep_poll = &param->nfc_dep_poll;
+                GASSERT(!nfc_dep_poll->g.bytes || nfc_dep_poll->g.size);
```

Comment:
I would drop this assert. Yes, empty memory blocks don't make much sense but technically speaking there's nothing wrong with it. And the code below handles empty memory blocks nicely AFAICT.